File: 509. Fibonacci Number.py
class Solution:
    def fib(self, n: int) -> int:
        
        ans = []
        if n==0:
            return 0
        elif n==1:
            return 1
        elif n==2:
            return 1
        else:
            
            ans.append(1)
            ans.append(1)
            
            for x in range(2,n+1):
                ans.append(ans[x-1]+ans[x-2])
            
            return ans[n-1]
        

File: cpp_solutions/509. Fibonacci Number.cpp
class Solution {
public:
    unordered_map<int, int> memo;
    int fib(int n) {
        
        if (n < 2){
            return n;
        }
        if (memo.find(n)!=memo.end()){
            return memo[n];
        }
        
        memo[n] = fib(n-1) + fib(n-2);
        return memo[n];
    }
};
